pkg/apis/resource/validation/validation.go

```diff
@@ -693,3 +711,51 @@ func validateMap[K ~string, T any](m map[K]T, maxSize int, validateKey func(K, *
 	}
 	return allErrs
 }
+
+func validateDeviceStatus(device resource.AllocatedDeviceStatus, fldPath *field.Path, allocatedDevices sets.Set[structured.DeviceID]) field.ErrorList {
+	var allErrs field.ErrorList
+	deviceID := structured.DeviceID{Driver: device.Driver, Pool: device.Pool, Device: device.Device}
+	if !allocatedDevices.Has(deviceID) {
+		allErrs = append(allErrs, field.Invalid(fldPath, deviceID, "must be an allocated device in the claim"))
+	}
+	allErrs = append(allErrs, validateSlice(device.Conditions, -1,
+		func(condition metav1.Condition, fldPath *field.Path) field.ErrorList {
+			var allErrs field.ErrorList
+			return allErrs
+		}, fldPath.Child("conditions"))...)
+	if len(device.Data.Raw) > 0 { // Data is an optional field.
+		allErrs = append(allErrs, validateRawExtension(device.Data, fldPath.Child("data"))...)
+	}
+	allErrs = append(allErrs, validateNetworkDeviceData(device.NetworkData, fldPath.Child("networkData"))...)
+	return allErrs
+}
+
+func validateRawExtension(rawExtension runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	// Validation of RawExtension as in https://github.com/kubernetes/kubernetes/pull/125549/
+	var v any
+	if len(rawExtension.Raw) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, ""))
+	} else if err := json.Unmarshal(rawExtension.Raw, &v); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, "<value omitted>", fmt.Sprintf("error parsing data: %v", err.Error())))
+	} else if v == nil {
+		allErrs = append(allErrs, field.Required(fldPath, ""))
+	} else if _, isObject := v.(map[string]any); !isObject {
+		allErrs = append(allErrs, field.Invalid(fldPath, "<value omitted>", "parameters must be a valid JSON object"))
+	}
+	return allErrs
+}
+
+func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fldPath *field.Path) field.ErrorList {
```

Comment:
I'm thinking we should validate the interface names
Should we apply the same validation than for linux interfaces? https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/utils.c?id=1f420318bda3cc62156e89e1b56d60cc744b48ad#n827

Reply:
I don't know if we need to do this - it's not going to work if the name is wrong

Reply:
fair

Reply:
I suggested in types.go that we do validate at least a length

Reply:
should we also validate that has the same characters as annotations per example, with the dns label validation regex IIRC?